src/main/drivers/gyro_sync.c

```diff
@@ -23,24 +22,35 @@ bool gyroSyncCheckUpdate(gyroDev_t *gyro)
     return gyro->intStatus(gyro);
 }
 
-uint32_t gyroSetSampleRate(uint8_t lpf, uint8_t gyroSyncDenominator)
+uint32_t gyroSetSampleRate(gyroDev_t *gyro, uint8_t lpf, uint8_t gyroSyncDenominator, bool gyro_use_32khz)
 {
+#ifndef GYRO_SUPPORTS_32KHZ
+    if (gyro_use_32khz) {
+        gyro_use_32khz = false;
+    }
+#endif
     int gyroSamplePeriod;
 
     if (lpf == GYRO_LPF_256HZ || lpf == GYRO_LPF_NONE) {
+        gyro->gyroRateKHz = GYRO_RATE_8_kHz;
         gyroSamplePeriod = 125;
     } else {
+        gyro->gyroRateKHz = GYRO_RATE_1_kHz;
         gyroSamplePeriod = 1000;
         gyroSyncDenominator = 1; // Always full Sampling 1khz
     }
 
     // calculate gyro divider and targetLooptime (expected cycleTime)
-    mpuDividerDrops  = gyroSyncDenominator - 1;
-    const uint32_t targetLooptime = gyroSyncDenominator * gyroSamplePeriod;
+    gyro->mpuDividerDrops  = gyroSyncDenominator - 1;
+    uint32_t targetLooptime = gyroSyncDenominator * gyroSamplePeriod;
+    if (gyro_use_32khz) {
+        gyro->gyroRateKHz = GYRO_RATE_32_kHz;
+        targetLooptime /= 4;
```

Comment:
This hardcoded dividing isn't great. The same calculation can be applied as new gyroSamplePeriod can be initialised as 32